### PR TITLE
support /pulp/deb proxying

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: manifests/plugin/deb.pp [hunk 1]
```diff
@@ -1,5 +1,33 @@
 # @summary Pulp Deb plugin
+# @param use_pulp2_content_route
 class pulpcore::plugin::deb {
```
Comment (ekohl) [inline manifests/plugin/deb.pp:3]:
```suggestion
class pulpcore::plugin::deb (
```

File: manifests/plugin/deb.pp [hunk 2]
```diff
@@ -1,5 +1,33 @@
 # @summary Pulp Deb plugin
+# @param use_pulp2_content_route
```
Comment (ekohl) [inline manifests/plugin/deb.pp:2]: Could you add a short description? It'll end up being rendered something like https://forge.puppet.com/theforeman/pulpcore/reference#pulpcorepluginrpm

**Review by ekohl — CHANGES_REQUESTED** [review]
Does this also work with Pulpcore 3.6 or is it 3.7 only? I'm trying to keep https://github.com/theforeman/puppet-pulpcore#support-policy correct.

File: manifests/plugin/deb.pp [hunk 3]
```diff
@@ -1,5 +1,33 @@
 # @summary Pulp Deb plugin
+# @param use_pulp2_content_route Whether to redirect the legacy (Pulp 2) URL, /pulp/deb/, to the content server
 class pulpcore::plugin::deb {
+  Boolean $use_pulp2_content_route = false,
+  } {
```
Comment (ekohl) [inline manifests/plugin/deb.pp:5]:
```suggestion
) {
```

**Review by ekohl — CHANGES_REQUESTED** [review]
You're using the wrong braces for parameters, causing it to fail.

File: manifests/plugin/deb.pp (hunk 3 above)
Comment (jlsherrill) [reply manifests/plugin/deb.pp:5]: oh shoot, sorry i got it backwards :)

File: manifests/plugin/deb.pp [hunk 4]
```diff
@@ -1,5 +1,33 @@
 # @summary Pulp Deb plugin
-class pulpcore::plugin::deb {
+# @param use_pulp2_content_route Whether to redirect the legacy (Pulp 2) URL, /pulp/deb/, to the content server
+class pulpcore::plugin::deb (
+  Boolean $use_pulp2_content_route = false,
+  ) {
```
Comment (ekohl) [inline manifests/plugin/deb.pp:5]: Lint is unhappy
```suggestion
) {
```

**Comment by jlsherrill** [comment]
```Does this also work with Pulpcore 3.6 or is it 3.7 only?```

It should work with both, nothing about it is specific to any pulpcore version

**Review by ekohl — APPROVED** [review]

**Pull request merged**